chore: add `pg_catalog` schema where needed

gbartolini opened the pull request:
Add the `pg_catalog` schema to fully qualify the usage of system functions and views provided by PostgreSQL.

gbartolini committed:
chore: fully qualify pg_catalog functions, views, etc.

Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

gbartolini commented:
/test

armru committed:
tests: adjust to the new queries

Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

pchovelon commented:
I see other files in which pg_catalog can be added, for example  :

- internal/management/controller/tablespaces/infrastructure/postgres.go
- internal/management/controller/tablespaces/infrastructure/postgres_test.go
- internal/management/controller/tablespaces/controller_test.go

gbartolini commented:
> * internal/management/controller/tablespaces/infrastructure/postgres.go

Definitely. Wanna give it a shot?

gbartolini committed:
Update pkg/management/postgres/logicalimport/roleinheritance_test.go

Co-authored-by: Pierrick <139142330+pchovelon@users.noreply.github.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

gbartolini committed:
Update pkg/management/postgres/logicalimport/roleinheritance.go

Co-authored-by: Pierrick <139142330+pchovelon@users.noreply.github.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

gbartolini committed:
Update pkg/management/postgres/logicalimport/role_test.go

Co-authored-by: Pierrick <139142330+pchovelon@users.noreply.github.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

gbartolini committed:
Update pkg/management/postgres/logicalimport/role_test.go

Co-authored-by: Pierrick <139142330+pchovelon@users.noreply.github.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

gabriele-wolfox commented:
We could also add pg_catalog here:

https://github.com/cloudnative-pg/cloudnative-pg/blob/main/tests/e2e/update_user_test.go#L167

`query := "SELECT rolpassword IS NULL FROM pg_authid WHERE rolname='postgres'"`

with

`query := "SELECT rolpassword IS NULL FROM [pg.catalog.pg](http://pg.catalog.pg/)_authid WHERE rolname='postgres'"`

https://github.com/cloudnative-pg/cloudnative-pg/blob/ebf36a59e54b5634b56ee6058d1b7551eafde083/tests/e2e/failover_test.go#L108

`query = fmt.Sprintf("SELECT pg_terminate_backend(pid) FROM pg_stat_replication "+`

with

`query = fmt.Sprintf("SELECT [pg.catalog.pg](http://pg.catalog.pg/)_terminate_backend(pid) FROM [pg.catalog.pg](http://pg.catalog.pg/)_stat_replication "+`

We could apply it also when `pg_stat_wal_receiver` occurs inside e2e.

gabriele-wolfox commented on pkg/management/postgres/logicalimport/role_test.go:
```diff
@@ -28,9 +28,9 @@ import (
-		"FROM pg_auth_members a LEFT JOIN pg_authid ur on ur.oid = a.roleid " +
-		"LEFT JOIN pg_authid um on um.oid = a.member " +
-		"LEFT JOIN pg_authid ug on ug.oid = a.grantor " +
+		"FROM pg_catalog.pg_auth_members a LEFT JOIN pg_authid ur on ur.oid = a.roleid " +
```
```suggestion
		"FROM pg_catalog.pg_auth_members a LEFT JOIN pg_catalog.pg_authid ur on ur.oid = a.roleid " +
```

This should fix the unit test failures

pchovelon commented:
> > * internal/management/controller/tablespaces/infrastructure/postgres.go
> 
> Definitely. Wanna give it a shot?

Yep, WIP.

armru committed:
fix: unit tests

Co-authored-by: Gabriele Quaresima <gabriele.quaresima@enterprisedb.com>

Co-authored-by: Gabriele Quaresima <gabriele.quaresima@enterprisedb.com>
Signed-off-by: Armando Ruocco <armando.ruocco.dev@gmail.com>
Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

mnencia committed:
Add more pg_catalog qualification to functions and tables

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

pchovelon committed:
chore: Adding pg_catalog in queries of tablespaces folder (#6797)

Linked to #6622

@gbartolini Don't know yet how to create a PR from my branch directly in
your dev/pg_catalog in a clean way. :/

Signed-off-by: Pierrick <pierrick.chovelon@dalibo.com>

mnencia committed:
chore: fix typo

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia commented:
/test tl=4

sxd commented on pkg/management/postgres/logicalimport/role.go:
```diff
@@ -166,8 +166,8 @@ func (rs *roleManager) getRoles(ctx context.Context) ([]Role, error) {
 			"pg_catalog.shobj_description(oid, 'pg_authid') as rolcomment, " +
-			"rolname = current_user AS is_current_user " +
-			"FROM pg_authid " +
+			"rolname = CURRENT_USER AS is_current_user " +
```
@mnencia shouldn't this be in lowercases?

sxd reviewed (commented):
Added one comment and triggered some e2e

sxd commented:
/test depth=push
